File: 백준/Gold/1253. 좋다/좋다.py
import sys
input = sys.stdin.readline

n = int(input())
num = list(map(int, input().split()))

def is_good_num(cur_num):
    num_list = num[:]
    num_list.pop(num_list.index(cur_num))
    left = 0
    right = len(num_list) - 1
    while left < right:
        two_num_sum = num_list[left] + num_list[right]

        if two_num_sum < cur_num:
            left += 1
        elif two_num_sum > cur_num:
            right -= 1
        else:
            return True
    return False

num.sort()

answer = 0
for i in range(n):
    if is_good_num(num[i]):
        answer += 1

print(answer)

File: 백준/Gold/1253. 좋다/좋다.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().split('\n')

const n = Number(input[0])
const arr = input[1].split(' ').map(Number)

arr.sort((a, b) => a - b)

let answer = 0
arr.forEach((num, index) => {
  let left = 0
  let right = n - 1

  while(left < right && right < n) {
    if (right === index) {
      right--
      continue
    }
    if (left === index) {
      left++
      continue
    }

    const sum = arr[left] + arr[right]

    if (num < sum) right--
    else if (sum < num) left++
    else {
      answer++
      break
    }
  }
})

console.log(answer)